sotodlib/io/bookbinder.py

```diff
@@ -1125,3 +1196,105 @@ def __call__(self):
 class TimingSystemError(Exception):
     """Exception raised when the timing system is on but timing counters are not found"""
     pass
+
+
+class HWPLoader:
+    """
+    Helper class to load in HWP data from data archive.
+    """
+    def __init__(self, root_dir):
+        self.root_dir = root_dir
+        self.files = []
+        self.data = {}
+        self.times = {} 
+
+        self.hwp_angle_interp = None
+        self.locked_interp = None
+        self.stable_interp = None
+        self.hwp_rate_interp = None
+
+    def get_files(self, start, stop):
+        """
+        Gets HWP files that may have data between start and stop time
+        """
+        files = []
+        for subdir in os.listdir(self.root_dir):
+            try:
+                tcode = int(subdir)
+            except:
+                continue
+
+            if not (((start//1e5) - 1) <= tcode <= stop//1e5):
```

Comment:
Why not stop//1e5 + 1?

Reply:
My reasoning for these start time buffers is just because the sample with timestamp `t` will fall in a file timestamped before the time `t`. So we have to check the file before the start time, but the file after the end time shouldn't contain any samples that we care about.

However as you mention below, the timestamp in the filename may not actually be a lower bound for the samples in file, so you are right that we should probably include a buffer on the `stop` time also.